### ISSUE TITLE
Support Screencast with audio

### ISSUE THREAD

**Issue by lkiesow** [elan-ev/opencast-studio#50]
Right now audio is included only in the webcam stream

**Comment by luniki** [comment]
I will have to investigate this. Is this possible?

**Comment by luniki** [comment]
Hmm, seems to be possible in Chrome >=v74

see:
https://developer.mozilla.org/en-US/docs/Web/API/Screen_Capture_API/Using_Screen_Capture#Capturing_shared_audio

**Comment by lkiesow** [comment]
Still does not work (Recorded in Google Chrome	78.0.3904.97 (Official Build) (64-bit)):
https://youtu.be/8wcytLaJuyg

Note that Chrome shows not up on the list of recording applications and the resulting video contains a video stream only:

```
Input #0, matroska,webm, from 'Presentation video - Recording (2).webm':
  Metadata:
    encoder         : Chrome
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv), 1678x800, SAR 1:1 DAR 839:400, 29.92 fps, 29.92 tbr, 1k tbn, 1k tbc (default)
    Metadata:
      alpha_mode      : 1

```

**Comment by luniki** [comment]
We just realized that Lars wants an audio track accompanying the screen recording **in case** that we do not record the camera.

**Comment by luniki** [comment]
I would like to postpone this issue until we have a concept for the "Choose your sources" UI. The effort to fix this twice should be avoided.

**Comment by lkiesow** [comment]
I need some estimates before we postpone this:

- how much time would you need to implement this?
- how would your re-design plan look like and much time would you need for that?
- how much of the code for just fixing the issue could you re-use? This functionality would stay after all.

**Comment by luniki** [comment]
>     * how much time would you need to implement this?

I can close this in KW 48 and will try to work on this today.

>     * how would your re-design plan look like and much time would you need for that?

I have to split the current "studio" page into "source setup" and "recording".

>     * how much of the code for just fixing the issue could you re-use? This functionality would stay after all.

There will be UI code. This cannot be shared. There will be code for setting up the audio recorder and managing several devices instead of the hard-coded two we currently use. This will probably be re-usable.

**Comment by LukasKalbertodt** [comment]
Fixed by #288
